app/scripts/services/user.js

```diff
@@ -31,39 +34,65 @@ angular.module('bballApp')
       }
     };
 
+    // backend serverAddr
+
     var currentUser = {};
 
-    // AngularJS will instantiate a singleton by calling "new" on this function
-    this.attemptRegister = function(username){
+    // helper function, returns user from users array
+    var getUser = function (users, username) {
+      return users[users.map(function (user) {
+        return user.username;
+      }).indexOf(username)];
+    };
+
+    this.register = function (username) {
+      var url = serverAddr + "users";
+      var promise = $http.get(url);
       var defer = $q.defer();
-      if (!username || username === undefined)
-        defer.reject("Username is not valid");
-      if (!users.hasOwnProperty(username.toLowerCase())){
-        users[username.toLowerCase()] = {name: username,
-          totalShots: 0,
-          highestStreak: 0,
-          shootOutsWon: 0};
-        currentUser = users[username.toLowerCase()];
-        $rootScope.userName = username;
-        defer.resolve(username);
-      }
-      else defer.reject("Username is already taken");
+
+      promise.then(function (getSuccessRes){
+        // if user already exists
+        if (getUser(getSuccessRes.data, username)){
+          defer.reject("Username is already taken");
+        } else { // username is available
+
+          var newUser = {
+            "username": username
+            //totalHoops: 0,
+            //highestStreak: 0,
+            //shootoutsWon: 0
+          };
+
+          var postRes = $http.post(
+            serverAddr + 'users',
+            JSON.stringify(newUser)
+          );
+          postRes.then(function(postSuccessRes){
+            defer.resolve(username);
+          }, function(postFailRes){
+            defer.reject("Post failed: " + postFailRes.statusText);
+          });
+        }
+      }, function (getFailRes) {
+        defer.reject("Get failed: " + getFailRes.statusText);
+      });
 
       return defer.promise;
     };
 
-    this.attemptLogin = function(username){
+    this.attemptLogin = function (username) {
       var defer = $q.defer();
-      if (!username)
+      if (!username) {
         defer.reject("Username is not valid");
-      if (users.hasOwnProperty(username.toLowerCase())){
+      } else if (users.hasOwnProperty(username.toLowerCase())) {
         currentUser = users[username.toLowerCase()];
-        $rootScope.userName = username;
         defer.resolve(username);
-      }
-      else defer.reject("Username is not registered");
+        $rootScope.userName = username;
```

Comment:
I had some difficulty difficulty getting the html to retrieve information from the user service so it was suggested that I put the information on rootScope instead, I have been trying to fix this but may need some assistance.